### PR TITLE
Add rfe/rfep/cmmb

### PR DESCRIPTION
**Pull request by YutHelloWorld**
Add 3 snippets for React Functional Component and Comments big block.
Inspired by [Our Best Practices for Writing React Components](https://engineering.musefind.com/our-best-practices-for-writing-react-components-dec3eb5c3fc8).

1.  rfe -> React functional export component
2.  rfep ->  React functional export component with propTypes
3. cmmb -> comment big block
4. modify code style: delete semi and add space before function parentheses inspired by [JavaScript Standard Style](https://standardjs.com/) 
5. change mapDispatchToProps as object. See [react-redux/mapDispatchToProps](https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options)
> [`mapDispatchToProps(dispatch, [ownProps]): dispatchProps`] (Object or Function): If an object is passed, each function inside it is assumed to be a Redux action creator. An object with the same function names, but with every action creator wrapped into a dispatch call so they may be invoked directly, will be merged into the component’s props.

Thank you for these great snippets!

### CODE REVIEW

File: snippets/snippets.json [hunk 1]
```diff
@@ -284,106 +283,142 @@
...
+      "export default ${1:componentName}",
       ""
     ],
     "description": "Creates a React component class with ES7 module system"
   },
+  "reactFunctionalExportComponent": {
+    "prefix": "rfe",
+    "body": [
+      "import React from 'react'",
+      "function ${1:componentName} () {",
```
Comment (dsznajder) [inline snippets/snippets.json]: Please add spaces after imports, also to keep consistency you can just write `const ComponentName = () =>`. I would like to keep this as arrow function. :)

File: README.md [hunk 2]
```diff
@@ -57,26 +62,29 @@ Every space inside `{ }` and `( )` means that this is pushed into next line :)
 |`cwup→`|`componentWillUpdate = (nextProps, nextState) => { }`|
 |`cdup→`|`componentDidUpdate = (prevProps, prevState) => { }`|
 |`cwun→`|`componentWillUnmount = () => { }`|
-|`ren→`|`render() { return( ) }`|
+|`ren→`|`render () { return( ) }`|
```
Comment (dsznajder) [inline README.md]: I'm not quite a fan of adding space after render, It's just a called function which is getting weird with something like. `this.functionName ()` than `this.functionName()`

Comment (YutHelloWorld) [reply README.md]: I‘m wrong. :). It should be as this:
```JS
render() {
  ...something
}

function MyComponent() {
   ...something
}
```
It should not put a space between function name and ( .

File: snippets/snippets.json (hunk 1 above)
Comment (YutHelloWorld) [reply snippets/snippets.json]:
```JS
const ExpandableForm = ({ onExpand, expanded, children }) => {
```
Looks very modern, but the function here is actually unnamed.
This lack of name will not be a problem if your Babel is set up correctly — but if it’s not, any errors will show up as occurring in <<anonymous>> which is terrible for debugging.

**Comment by YutHelloWorld** [comment]
@dsznajder  I fix some thing

File: README.md [hunk 3]
```diff
@@ -123,40 +132,46 @@ Every space inside `{ }` and `( )` means that this is pushed into next line :)
...
+
+### `rce`
+
 ```javascript
-import React, { Component } from 'react';
+import React, { Component } from 'react'
 
 export class $1 extends Component {
-  render() {
+  render () {
```
Comment (dsznajder) [inline README.md]: ✂️  Space :D

File: snippets/snippets.json [hunk 4]
```diff
@@ -535,25 +573,25 @@
    "componentRender": {
     "prefix": "ren",
     "body": [
-      "render() {",
+      "render () {",
```
Comment (dsznajder) [inline snippets/snippets.json]: Space :scissors:

File: snippets/snippets.json [hunk 5]
```diff
@@ -617,37 +655,37 @@
...
     "prefix": "rnc",
     "body": [
-      "import React, { Component } from 'react';",
-      "import { Text, View } from 'react-native';",
+      "import React, { Component } from 'react'",
+      "import { Text, View } from 'react-native'",
       "",
       "export default class ${1:componentName} extends Component {",
-      "  render() {",
+      "  render () {",
```
Comment (dsznajder) [inline snippets/snippets.json]: Space :scissors:

File: snippets/snippets.json [hunk 6]
```diff
@@ -617,37 +655,37 @@
...
     "prefix": "rnce",
     "body": [
-      "import React, { Component } from 'react';",
-      "import { Text, View } from 'react-native';",
+      "import React, { Component } from 'react'",
+      "import { Text, View } from 'react-native'",
       "",
       "export class ${1:componentName} extends Component {",
-      "  render() {",
+      "  render () {",
```
Comment (dsznajder) [inline snippets/snippets.json]: Space :scissors:

File: snippets/snippets.json [hunk 7]
```diff
@@ -284,28 +283,68 @@
...
+    "prefix": "rfep",
+    "body": [
+      "import React from 'react'",
+      "import PropTypes from 'prop-types'",
+      "",
+      "${1:componentName}.propTypes = {",
+      "",
+      "}",
+      "",
+      "function ${1:componentName} () {",
```
Comment (dsznajder) [inline snippets/snippets.json]: Space after the name of a function is correct for you? :) Still, I'm not a fan of `function` :).

File: snippets/snippets.json (hunk 6 above)
Comment (YutHelloWorld) [reply snippets/snippets.json]: fixed these 😂

**Comment by dsznajder** [comment]
Ok :) @YutHelloWorld Nice work :) Thanks for contribution

**Comment by YutHelloWorld** [comment]
the stateless functional components is not in readme, maybe you need add them

**Pull request merged**